### PR TITLE
Refactor structure of mmrotate

### PR DESCRIPTION
**Pull request by zytx121**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Refactor structure of mmrotate.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmrotate/datasets/pipelines/__init__.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 # Copyright (c) OpenMMLab. All rights reserved.
-from .loading import LoadPatchFromImage
-from .transforms import PolyRandomRotate, RMosaic, RRandomFlip, RResize
+from .loading import LoadPatchfromNDArray
+from .transforms import (ConvertBoxType, RandomChoiceRotate, RandomRotate,
+                         Rotate)
```
Comment (RangiLyu) [inline mmrotate/datasets/pipelines/__init__.py:4]: mmrotate/datasets/pipelines -> mmrotate/datasets/transforms

File: mmrotate/evaluation/functional/bbox_overlaps.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from mmcv.ops import box_iou_rotated
+from torch import Tensor
+
+from mmrotate.structures.bbox import RotatedBoxes
+
+
+def rbbox_overlaps(bboxes1: Tensor,
```
Comment (RangiLyu) [inline mmrotate/evaluation/functional/bbox_overlaps.py:8]: the bbox_overlaps in mmrotate/evaluation/functional/bbox_overlaps.py should be numpy version, the tensor version should be put in mmrotate/structures/bbox/bbox_overlaps.py

File: mmrotate/models/task_modules/__init__.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from .align import FRM, AlignConv, DCNAlignModule, PseudoAlignModule
```
Comment (RangiLyu) [inline mmrotate/models/task_modules/__init__.py:2]: better to move them to mmrotate/models/layers

**Review by RangiLyu — APPROVED** [review]
LGTM

**Comment by l4d2boomer** [comment]
@zytx121 I am wondering why almost all files in the detector folder have deleted. 😭

**Pull request merged**